librz/core/cprint.c

```diff
@@ -487,6 +487,31 @@ RZ_IPI bool rz_core_print_hexdump_byline(RZ_NONNULL RzCore *core, bool hexoffset
 	return true;
 }
 
+/**
+ * \brief Hexdump containing references at \p addr
+ * \param address Dump bytes address
+ * \param len Dump bytes length
+ * \return Hexdump string
+ */
+RZ_IPI RZ_OWN char *rz_core_print_hexdump_refs(RZ_NONNULL RzCore *core, ut64 address, size_t len, int wordsize) {
+	rz_return_val_if_fail(core, NULL);
+	const int ocols = core->print->cols;
+	int bitsize = core->rasm->bits;
+	/* Thumb is 16bit arm but handles 32bit data */
+	if (bitsize == 16) {
+		bitsize = 32;
+	}
+	core->print->cols = 1;
+	core->print->flags |= RZ_PRINT_FLAGS_REFS;
+	// TODO: Do not use core->block and switch to using proper API:
+	// https://github.com/rizinorg/rizin/issues/2695
+	char *hexdump_str = rz_print_hexdump_str(core->print, address, core->block,
+		RZ_MIN(len, core->blocksize), wordsize * 8, bitsize / 8, 1);
```

Comment:
Shouldn't we use rz_io_read here instead of core->block? I think there's the risk that we print the wrong content otherwise.

Reply:
oh i missed this. i agree also.